builtin/v9/migration/top.go

```diff
@@ -135,30 +156,34 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, newManifestCID
 		return cid.Undef, xerrors.Errorf("didn't find miner in old manifest entries")
 	}
 
-	for name, oldCodeCID := range simpleMigrations { //nolint:nomaprange
-		newCodeCID, ok := newManifest.Get(name)
-		if !ok {
-			return cid.Undef, xerrors.Errorf("code cid for %s actor not found in new manifest", name)
-		}
+	for name, oldCodeCID := range oldCodeIDMap { //nolint:nomaprange
+		if name == manifest.MarketKey || name == manifest.VerifregKey {
+			deferredCodeIDs[oldCodeCID] = struct{}{}
+		} else {
+			newCodeCID, ok := newManifest.Get(name)
+			if !ok {
+				return cid.Undef, xerrors.Errorf("code cid for %s actor not found in new manifest", name)
+			}
 
-		migrations[oldCodeCID] = codeMigrator{newCodeCID}
+			migrations[oldCodeCID] = codeMigrator{newCodeCID}
+		}
 	}
 
-	// migrations that migrate both code and state
+	// migrations that migrate both code and state, override entries in `migrations`
+
+	// The System Actor
+
 	newSystemCodeCID, ok := newManifest.Get("system")
 	if !ok {
 		return cid.Undef, xerrors.Errorf("code cid for system actor not found in manifest")
 	}
 
-	miner9Cid, ok := newManifest.Get("storageminer")
-	if !ok {
-		return cid.Undef, xerrors.Errorf("code cid for miner actor not found in new manifest")
-	}
-
 	migrations[systemActor.Code] = systemActorMigrator{newSystemCodeCID, newManifest.Data}
 
+	// The Miner Actor
```

Comment:
nit: move comment to code it refers to

Reply:
This was intentionally placed, because the market state loading is for The Miner Actor. I can move it if it's only serving to confuse, though.

Reply:
ah ok in that case writing a bit more would also do it

Reply:
Done